Apromore-Extras/APMLogModule/src/main/java/org/apromore/apmlog/DefaultChartDataCollection.java

```diff
@@ -189,125 +189,139 @@ public DefaultChartDataCollection(APMLog apmLog) {
+                Collections.sort(allAverageWaitingTimes);
+                int numOfUnits = MAX_UNIT;
+                double first = allAverageWaitingTimes.get(0);
+                double last = allAverageWaitingTimes.get(allAverageWaitingTimes.size() - 1);
+                avgWTUnit = (last - first) / numOfUnits;
+                double current = first - avgWTUnit;
+                for (int i = 0; i <= numOfUnits; i++) {
+                    if (i == numOfUnits) current = last;
+                    else current += avgWTUnit;
+                    avgWTMap.put(current, 0);
+                }
             }
         }
         /**
          * For max waiting time
          */
         TreeSortedMap<Double, Integer> maxWTMap = new TreeSortedMap<>();
         double maxWTUnit = 0;
-        if(interval > 0) {
-            Collections.sort(allMaxWaitingTimes);
-            int numOfUnits = MAX_UNIT;
-            double first = allMaxWaitingTimes.get(0);
-            double last = allMaxWaitingTimes.get(allMaxWaitingTimes.size() - 1);
-            maxWTUnit = (last - first) / numOfUnits;
-            double current = first - maxWTUnit;
-            for(int i=0; i <= numOfUnits; i++) {
-                if(i == numOfUnits) current = last;
-                else current += maxWTUnit;
-                maxWTMap.put(current, 0);
+        if (allMaxWaitingTimes.size() > 0 ) {
+            if (interval > 0) {
+                Collections.sort(allMaxWaitingTimes);
+                int numOfUnits = MAX_UNIT;
+                double first = allMaxWaitingTimes.get(0);
+                double last = allMaxWaitingTimes.get(allMaxWaitingTimes.size() - 1);
+                maxWTUnit = (last - first) / numOfUnits;
+                double current = first - maxWTUnit;
+                for (int i = 0; i <= numOfUnits; i++) {
+                    if (i == numOfUnits) current = last;
+                    else current += maxWTUnit;
+                    maxWTMap.put(current, 0);
+                }
             }
         }
         /**
          * For case utilization
          */
         TreeSortedMap<Double, Integer> caseUtilMap = new TreeSortedMap<>();
         double caseUtilUnit = 0;
-        if(interval > 0) {
-            Collections.sort(allCaseUtils);
-            int numOfUnits = (MAX_UNIT / 2);
-            double first = allCaseUtils.get(0);
-            double last = allCaseUtils.get(allCaseUtils.size() - 1);
-            caseUtilUnit = (last - first) / numOfUnits;
-            double current = first - caseUtilUnit;
-            for(int i=0; i < numOfUnits; i++) {
-                current += caseUtilUnit;
-                caseUtilMap.put(current, 0);
+        if (allCaseUtils.size() > 0 ) {
+            if (interval > 0) {
+                Collections.sort(allCaseUtils);
+                int numOfUnits = (MAX_UNIT / 2);
+                double first = allCaseUtils.get(0);
+                double last = allCaseUtils.get(allCaseUtils.size() - 1);
```

Comment:
Same as above

Reply:
This class will be removed later. So, I'm not going to change the code.

Reply:
But this is new code that is added in the class. 
Il make the changes, till then you can work on other stuff in the same